opm/autodiff/MultisegmentWell_impl.hpp

```diff
@@ -1224,9 +1224,9 @@ namespace Opm
 
             segment_viscosities_[seg] = 0.;
             // calculate the average viscosity
-            for (int p = 0; p < number_of_phases_; ++p) {
-                const EvalWell phase_fraction = mix[p] / b[p] / volrat;
-                segment_viscosities_[seg] += visc[p] * phase_fraction;
+            for (int comp_idx = 0; comp_idx < num_components_; ++comp_idx) {
+                const EvalWell comp_fraction = mix[comp_idx] / b[comp_idx] / volrat;
+                segment_viscosities_[seg] += visc[comp_idx] * comp_fraction;
```

Comment:
I am a little confused as to this change: viscosity is associated with phases, not components. I guess I don't know if the mix is by phase or components either...

Reply:
We want to sum the viscosities of all the components. The size of visc (and mix and b) is num_components_ not number_of_phases_.  If solvent is present (should be included sometime) , it should be added to this sum.